### DISCUSSION TITLE
Integrate Foundry

### DISCUSSION THREAD

**Discussion by Cr0wn-Gh0ul** [alicenet/alicenet#502] [Ideas]
I would like for us to integrate Foundry into AliceNet for smart contract testing as this will enable us to do better unit testing and add fuzzing capibilites. Information for foundry can be found [here](https://www.paradigm.xyz/2021/12/introducing-the-foundry-ethereum-development-toolbox). I also have made an example template repo: [Smart-Contract-Base](https://github.com/Cr0wn-Gh0ul/smart-contract-base), which shows how hardhat and foundry can be used together.

**Comment by ghost** [comment]
Just checked and it supports code coverage which I’m a fan of. A full migration to this and we can rip out all of our Javascript tooling in this repo.

**Reply by Cr0wn-Gh0ul** [reply]
I'm not sure we will want to do a full migration, at least not initially. We would want to bring it along side hardhat just to improve testing capabilities by adding fuzzing. Maybe a transition later down the line as time allows, but we don't have the luxury of time at the moment.

**Reply by Cr0wn-Gh0ul** [reply]
Along that note this inspiration comes from [SeaPort](https://github.com/ProjectOpenSea/seaport)
